syntax = "proto3";

package tc_pb;

import "transport_router.proto";
import "map_renderer.proto";




enum BusType {
    ORDINARY = 0;
    CYCLED = 1;
}

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    int32 id = 1;
    string name = 2;
    Coordinates coordinates = 3;
} 

message Bus {
    int32 id = 1;
    string name = 2;
    repeated int32 stops = 3;
    bool bus_type_cycled = 4;    
}

message StopToBuses {
    int32 stop_id = 1;
    repeated int32 bus_id = 2;
}

message IntervalToDistance {
    int32 from_id = 1;
    int32 to_id = 2;
    int64 distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated StopToBuses stops_to_buses = 3;
    repeated IntervalToDistance intervals_to_distance = 4;
}

message TransportBase {
    TransportCatalogue cat = 1;
    RoutingSettings routing_settings = 2;
    RenderSettings rendder_settings = 3;

    TransportRouter transport_router = 4;
    Router router = 5;
}



